addons/udes_stock/models/stock_picking.py

```diff
@@ -465,8 +465,17 @@ def button_draft(self):
         when the button is clicked.
         Normally should not be any move lines, adding extra safety just in case in an
         unknown scenario there are move lines.
+        Additionally, Ensures that the Delivery Control process is completed before setting the picking to draft. 
+        This is triggered if there is a linked delivery control and its state is not "done".
         """
         self.ensure_one()
+
+        if (
+            self.u_delivery_control_picking_id
+            and self.u_delivery_control_picking_id.state != "done"
+        ):
+            raise UserError("Complete the delivery control process first.")
```

Comment:
Better to provide the delivery control name also in the error message, that would be helpful to users.